### PR TITLE
improve terrahub speed

### LINKED ISSUE #333: Investigate and if possible improve terrahub command speed by ignoring node_modules or .terraform folders
## Describe the Feature

Investigate and if possible improve terrahub command speed by ignoring node_modules or .terraform folders

## Expected Behavior

Investigate and if possible improve terrahub command speed by ignoring node_modules or .terraform folders

### PR DESCRIPTION
**Pull request by KanoniPan**
## Description

improve terrahub speed
Fixes #333 

## Type of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a change to the documentation

## How Has This Been Tested?

- `terrahub init`

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have checked that my changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/templates/config/project/.terrahub.yml.twig [hunk 1]
```diff
@@ -3,6 +3,11 @@ project:
   name: '{{ name }}'
   code: '{{ code }}'
   provider: '{{ provider }}'
+  include:
```
Comment (eistrati) [inline src/templates/config/project/.terrahub.yml.twig]: add this config into `src/templates/config/project/.terrahub.json.twig` as well

**Comment by KanoniPan** [comment]
@eistrati  to search configs `terrahub` uses [glob](https://www.npmjs.com/package/glob)
so i propose to use the following template

  ```yaml
include:
   - ./
  ignore:
   - **/.terraform/*
   - **/node_modules/*
```
note: include folders can be given only directly without regex or glob (relative or absolute path)

File: src/templates/config/project/.terrahub.json.twig [hunk 2]
```diff
@@ -3,6 +3,11 @@
     "name": "{{ name }}",
     "code": "{{ code }}",
     "provider": "{{ provider }}"
```
Comment (eistrati) [inline src/templates/config/project/.terrahub.json.twig]: I don't believe this JSON will be valid. Please review and fix it.

File: src/config-loader.js [hunk 3]
```diff
@@ -256,11 +264,12 @@ class ConfigLoader {
...
    * @param {String} pattern
    * @param {String} path
+   * @param {String[]} ignore
    * @returns {*}
    * @private
    */
-  _find(pattern, path) {
-    return glob.sync(pattern, { cwd: path, absolute: true, dot: true, ignore: ConfigLoader.IGNORE_PATTERNS });
+  _find(pattern, path, ignore = []) {
+    return glob.sync(pattern, { cwd: path, absolute: true, dot: true, ignore: ConfigLoader.IGNORE_PATTERNS.concat(ignore) });
```
Comment (Leichtwind) [inline src/config-loader.js]: change `IGNORE_PATTERNS` `get` method to
```
return this.getProjectConfig().ignore || [];
```
and remove `ingore` from method args

File: src/config-loader.js [hunk 4]
```diff
@@ -133,7 +133,15 @@ class ConfigLoader {
...
-
+    const { include, ignore } = this.getProjectConfig();
+    if (include && include.length) {
+      const ignoreArray = []
+
+      include.forEach(it => {
+        ignoreArray.push(...this._find('**/.terrahub.+(json|yml|yaml)', path.resolve(this.appPath(), it), ignore));
+      })
+      return ignoreArray
+    }
```
Comment (Leichtwind) [inline src/config-loader.js]: simplify this construction by
```js
let searchParhs;
if (dir) {
  searchPaths = dir;
} else if (include && include.length) {
  searchPaths = ignore.map(it => path.resolve(this.appPath(), it));
} else {
  searchPaths = this.appPath()
}
```

File: src/templates/config/project/.terrahub.yml.twig (hunk 1 above)
Comment (Leichtwind) [reply src/templates/config/project/.terrahub.yml.twig]: `include` and `ignore` were added in `.terrahub.json.twig`

**Pull request merged**